ReactAndroid/build.gradle

```diff
@@ -133,21 +133,14 @@ task prepareGlog(dependsOn: dependenciesPath ? [] : [downloadGlog], type: Copy)
     }
 }
 
-task downloadJSC(dependsOn: createNativeDepsDirectories, type: Download) {
-    src("https://registry.npmjs.org/jsc-android/-/jsc-android-${JSC_VERSION}.tgz")
-    onlyIfNewer(true)
-    overwrite(false)
-    dest(new File(downloadsDir, "jsc-${JSC_VERSION}.tar.gz"))
-}
-
 // Create Android.mk library module based on jsc from npm
-task prepareJSC(dependsOn: downloadJSC) {
```

Comment:
Bit confused about removing downloadJSC dependency.

What makes it appear jsc-android at `"$projectDir/../node_modules/jsc-android/dist"` location by the time this script runs? 

FYI I don't see build.gradle invoking npm at any point, so unless I'm missing it somewhere this may make builds without invoking `npm install` first always fail.

Reply:
This is inside of `node_modules/react-native` when using React Native, so JSC should definitely be there.

Reply:
I meant when building RN from source, not when consuming it. While consuming none of ReactAndroid/build.gradle matters anyways.

The current instructions to build from source is simple download RN and run `./gradlew :ReactAndroid:installArchives` I am asking will this work?

Reply:
Good point. Is there a way we can invoke "yarn" from gradle just to make sure node modules are in sync?

Reply:
A comment from when new JSC was checked in https://github.com/facebook/react-native/pull/22263#issuecomment-441437249

> Using npm is just not the right way to manage jsc as it's a native dependency. After fighting with it for a while I abandoned this approach. The correct approach (which will require some work in jsc-android-buildscripts) is to manage publications of jsc in maven. This way the user will be able to override / use other versions of jsc.

Reply:
cc @DanielZlotin

Reply:
From my point of view, RN itself is tightly coupled with node_modules, i.e. you cannot simply get a AAR or APK to run without metro or node_modules stuffs.

I also prefer to maintain even native dependencies in single source of truth, the pakcage.json.
Otherwise, we currently have multiple place to declare dependency version.
E.g. folly version 2018.10.22.00 defines in multiple places like gradle, CocoaPods, and Xcodeproj.

Invoking yarn by gradle is good to me.
Or at least we could check if node_modules is not there and prompt user to run yarn manually.

Reply:
Note this also broke the docker build `yarn test-android-unit`. I tried changing the `Dockerfile.android` to `RUN yarn` before `./gradlew :ReactAndroid:packageReactNdkLibsForBuck` (and removed the dead reference to `downloadJSC`) but that didn't work, it still failed with
```
> Expected directory '/app/ReactAndroid/../node_modules/jsc-android/dist' to contain exactly one file, however, it contains no files.
```